polykit/renderers/viewers.py

```diff
@@ -63,3 +66,90 @@ def chromosome_viewer(chrom_lengths,
         cb.update_ticks()
         
     plt.show()
+
+
+def rasterize(positions,
+              box_size,
+              resolution=100,
+              length_unit=50,
+              gaussian_width=250):
+    """
+    3D positions rasterizer for numerical microscopy analysis
+    
+    Parameters
+        ----------
+        positions : Nx3 float array
+            List of X,Y,Z particle positions to be rasterize.
+            Assumes particle coordinates are wrapped in PBCs within the range [-box_size/2,+box_size/2]
+        box_size : float
+            Linear dimension of periodic box
+        resolution : float
+			Linear dimension of output voxels (in nm)
+        length_unit : float
+            Model unit of length (in nm)
+        gaussian_width : float
+            Width of Gaussian point-spread function to be used.
+            If gaussian_width<=0, returns the raw (undiffracted) raster
+    """
+    
+    n_voxels = int(box_size / (resolution/length_unit))
+    bins = np.linspace(-box_size/2., box_size/2., num=n_voxels+1)
+																			
+    digitized_x = np.digitize(positions[:,0], bins) - 1
+    digitized_y = np.digitize(positions[:,1], bins) - 1
+    digitized_z = np.digitize(positions[:,2], bins) - 1
+					
+    digitized = digitized_x + digitized_y*n_voxels + digitized_z*n_voxels**2
+    bincounts = np.bincount(digitized, minlength=n_voxels**3)
+					
+    raster = bincounts.reshape((n_voxels,n_voxels,n_voxels))
+	
+    if gaussian_width > 0:
+        smoothed_raster = gaussian_filter(raster/raster.max(), sigma=gaussian_width/resolution, mode='wrap')
+
+        return smoothed_raster
+        
+    return raster/raster.max()
```

Comment:
would it make sense to offer returning raw counts per bin too ? e.g. if max_scaled: return raster/raster.max() else: return raster ?